receiver/mongodbatlasreceiver/internal/mongodb_atlas_client.go

```diff
@@ -25,6 +27,38 @@ import (
 	"go.uber.org/zap"
 )
 
+type RetryBackoffRoundTripper struct {
+	originalTransport http.RoundTripper
+	log               *zap.Logger
+	Attempts          int
+	RetryDelay        int // In ms
+}
+
+func (rt *RetryBackoffRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+
+	resp, err := rt.originalTransport.RoundTrip(r)
+	if err != nil {
+		return nil, err // Can't do anything
+	}
+	if resp.StatusCode == 429 {
+		for i := 0; i < rt.Attempts; i++ {
+			delay := rt.RetryDelay << i
+			time.Sleep(time.Duration(delay) * time.Millisecond)
+			rt.log.Warn("server busy, retrying request",
+				zap.Int("attempts", i+1),
+				zap.Int("delay", delay))
+			resp, err = rt.originalTransport.RoundTrip(r)
+			if err != nil {
+				return nil, err
+			}
+			if resp.StatusCode != 429 {
+				break
+			}
+		}
+	}
+	return resp, err
+}
```

Comment:
would be interested to hear about the use-case, especially because we will re-scrape anyway in collection interval.

Reply:
The limit is somewhat low for the Atlas service, only [100 requests/min/project](https://docs.cloudmanager.mongodb.com/core/api/#automation). A company with a large cluster appears to be hitting that limit with every scrape, so they never get data out of this scraper.

Reply:
But by doing more requests a.k.a scrapes via retry you don't help correct?

Reply:
Each scrape has many requests (it's a really talkative API), so this PR allows for backoff _during_ a scrape, allowing the scrape to complete.